resource-validator/src/test/scala/com/broadinstitute/dsp/resourceValidator/InitBucketCheckerSpec.scala

```diff
@@ -0,0 +1,63 @@
+package com.broadinstitute.dsp
+package resourceValidator
+
+
+import cats.effect.IO
+import com.broadinstitute.dsp.Generators._
+import com.google.cloud.storage.{Bucket, Storage}
+import fs2.Stream
+import org.broadinstitute.dsde.workbench.RetryConfig
+import org.broadinstitute.dsde.workbench.google2.mock.{BaseFakeGoogleStorage, FakeGoogleStorageInterpreter}
+import org.broadinstitute.dsde.workbench.google2.GoogleStorageService
+import org.broadinstitute.dsde.workbench.model.TraceId
+import org.broadinstitute.dsde.workbench.model.google.{GcsBucketName, GoogleProject}
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatestplus.mockito.MockitoSugar
+
+class InitBucketCheckerSpec extends AnyFlatSpec with CronJobsTestSuite with MockitoSugar {
+    val mockBucket = mock[Bucket]
+
+    it should "return None if bucket no longer exists in Google" in {
+      val storageService = new BaseFakeGoogleStorage {
+        override def getBucket(googleProject: GoogleProject, bucketName: GcsBucketName, bucketGetOptions: List[Storage.BucketGetOption], traceId: Option[TraceId]): IO[Option[Bucket]] =
+          IO.pure(None)
+      }
+      val checkerDeps =
+        initBucketCheckerDeps(storageService)
+
+      forAll { (bucket: InitBucketToRemove, dryRun: Boolean) =>
+        val dbReader = new FakeDbReader {
+          override def getInitBucketsToDelete: Stream[IO, InitBucketToRemove] = Stream.emit(bucket)
+        }
+        val checker = InitBucketChecker.impl(dbReader, checkerDeps)
+        val res = checker.checkResource(bucket, dryRun)
+        res.unsafeRunSync() shouldBe None
+      }
+    }
+
+    it should "return bucket if bucket still exists in Google" in {
+      forAll { (bucket: InitBucketToRemove, dryRun: Boolean) =>
+        val storageService = new BaseFakeGoogleStorage {
+          override def getBucket(googleProject: GoogleProject, bucketName: GcsBucketName, bucketGetOptions: List[Storage.BucketGetOption], traceId: Option[TraceId]): IO[Option[Bucket]] =
```

Comment:
I don't think we're calling `getBucket` anywhere in actual code, no? so I think we can remove mockito as well

Reply:
We use it here https://github.com/broadinstitute/leonardo-cron-jobs/pull/14/files#diff-7de04ba3526138e76c3434cbbf7b17a7a85d86df137ad6a7635fb1a3fe3d4012R33 .  We don't do this for the current BucketRemover, since its not an anomaly and blindly calls delete and then reacts based on the  true/false response. In this case, we can't blindly issue the call because we want to support dryRun, which needs to know whether the bucket exists without calling delete first.